aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +23,51 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    int64_t self_num_batch_dims) {
+  // This implementation only applies to the batched grad case
+  auto other_sizes = other.sizes();
+  auto other_strides = other.strides();
+  auto other_storage_offset = other.storage_offset();
+
+  auto self_sizes = self.sizes();
+  auto self_strides = self.strides();
+  auto self_num_feature_dims = self.dim() - self_num_batch_dims;
+
+  // NB: We don't check this because we may allow broadcasting
+  // for (int i = 0; i < other.dim(); ++i) {
+  //   TORCH_CHECK(
+  //       other_sizes[i] == self_sizes[i + self_num_batch_dims],
+  //       "Expected the size of self at dim to be equivalent to the size of other at dim ", i);
+  // }
+
+  std::vector<int64_t> out_sizes;
+  out_sizes.reserve(self.dim());
+  out_sizes.insert(out_sizes.begin(), other_sizes.begin(), other_sizes.end());
+  out_sizes.insert(out_sizes.begin(), self_sizes.begin(), self_sizes.end() - self_num_feature_dims);
+
+  // We use the strides of other, and tack on the strides computed with
+  // the batch dims of self, so that the slices are arranged contiguously
+  std::vector<int64_t> out_strides;
+  out_strides.reserve(self.dim());
+  out_strides.insert(out_strides.begin(), other_strides.begin(), other_strides.end());
+
+  int64_t prod = native::storage_size_for(other_sizes, other_strides) + other_storage_offset;
```

Comment:
That's not correct? What if you have stuff after as well?
If you just want to access the number of element of the storage, we should do that.

Reply:
Corrected!